docs/dev/guidelines/database.rst

```diff
@@ -0,0 +1,236 @@
+**********************
+Database Relationship
+**********************
+
+WORK IN PROGRESS
+
+1. Retrieving and Building Objects
+==================================
+
+The cost of retrieving and building an object's relationships far exceeds the cost of selecting the object. This is especially true for relationships where it would trigger the loading of every child through the relationship hierarchy. The solution to this issue is **lazy fetching** (lazy loading). Lazy fetching allows the fetching of a relationship to be deferred until it is accessed. This is important not only to avoid the database access, but also to avoid the cost of building the objects if they are not needed. |br|
```

Comment:
what do you mean with retrieving and building an object's relationship?

should we use database language, like fetching an object vs joining tables?

Reply:
Retrieving and building refers to creating the java object and instantiating its relationships to other objects. Selecting the object refers to simply selecting an object from the database. I'm not sure how to reformulate this more clearly. Do you have a suggestion?